### ISSUE TITLE
[PhoneInput] typescript type and flow type not matching

### ISSUE THREAD

**Issue by ebati** [uber/baseweb#2547]
## Current Behavior

[Flow type](https://github.com/uber/baseweb/blob/bc4e68595c81b7d872896976b46952c4e0d38f48/src/phone-input/types.js#L87) is correct but [typescript type](https://github.com/uber/baseweb/blob/bc4e68595c81b7d872896976b46952c4e0d38f48/src/phone-input/index.d.ts#L325) missing some props.

## Expected Behavior

Both should match

## Your Environment

| Tech    | Version |
| ------- | ------- |
| Base UI | v9.34.0  |

**Comment by gergelyke** [comment]
@ebati thank you for opening this bugreport! could you help us sending a PR to fix it?

**Comment by ebati** [comment]
I tried to update typescript type definition. But i need some help.

Type definition in Flow show requirement of at least listed properties but for typescript type definition shows exact type.

Since the extra fields in phone-input is forwarded to input, there need to be a relation between phone-input props and input props, or combination of all props need to be listed under phone-input. How should i express this relation?

**Comment by gergelyke** [comment]
@chasestarr @tajo can you help here? sadly, I am no flow or typescript expert

**Comment by tajo** [comment]
@ebati Can we import `BaseInputProps` type from input and use that for phone input types?

**Comment by andreimc** [comment]
https://github.com/uber/baseweb/pull/2784

**Comment by andreimc** [comment]
this should fix the typescript errors, as a PhoneInput is an input we do the following a a workaround but PR is up

```type BasePhoneProps = PhoneInputProps & InputProps
const BasePhoneInput = PhoneInput as React.FC<BasePhoneProps>```

**Comment by chasestarr** [comment]
fixed in #2784
